majit/majit-ir/src/intbound.rs

```
@@ -0,0 +1,2611 @@
+
+    /// Return true if the sets of numbers self and other must be disjoint.
+    pub fn known_ne(&self, other: &IntBound) -> bool {
+        if self.known_lt(other) || self.known_gt(other) {
+            return true;
+        }
+        let both_known = self.tmask | other.tmask;
+        if unmask_zero(self.tvalue, both_known) != unmask_zero(other.tvalue, both_known) {
+            return true;
+        }
+        let mut newself = self.clone();
+        match newself.intersect(other) {
```

Clippy (1.95.0)

[warning] redundant pattern matching, consider using `is_err()`

(warning)